src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -1874,6 +1886,9 @@ internal async Task<List<VariableInformation>> GetLocalsAndParameters(AD7Thread
                 variables.Add(vi);
             }
 
+            if (ReturnValue != null)
```

Comment:
Wait, we are missing a check here: we only want to do this if we are on the top frame of the thread that stopped.